### PR TITLE
Droplet dynamic property block dragging

### PR DESCRIPTION
**Pull request by cpirich**
- Includes `expand_expansion` branch changes proposed for `droplet-editor` repo: see https://github.com/droplet-editor/droplet/compare/code-dot-org...expand_expansion
- Introduces the notion of the type `readonlyproperty` in dropletConfig (in addition to the typical `property`)
- For all blocks of type `property`, we now generate an `expansion` function such that the block can morph between being a getter or a setter depending on where the block is dropped. (The droplet changes above make it possible to do this with an `expansion` function)
  dropletproperties

### CODE REVIEW

**Comment by cpirich** [comment]
@Bjvanminnen @joshlory

**Comment by Bjvanminnen** [comment]
Based on screencast, this looks pretty cool :)

File: apps/src/dropletUtils.js [hunk 1]
```diff
@@ -311,6 +322,12 @@ exports.generateDropletPalette = function (codeFunctions, dropletConfig) {
       }
     }
 
+    // For properties that aren't read-only, we automatically generate an
+    // expansion function so the block can morph to be a setter and a getter:
+    if (!expansion && funcInfo.type === 'property') {
+      expansion = generatePropertyExpansion(block);
+    }
```
Comment (Bjvanminnen) [inline apps/src/dropletUtils.js]: Where is expansion initially declared?

File: apps/src/dropletUtils.js [hunk 2]
```diff
@@ -482,14 +499,19 @@ function getModeOptionFunctionsFromConfig(config) {
...
+          break;
+        case 'either':
+          newFunc.value = true;
+          newFunc.command = true;
+          break;
+        case 'property':
+        case 'readonlyproperty':
+          newFunc.property = true;
+          newFunc.value = true;
+          break;
```
Comment (Bjvanminnen) [inline apps/src/dropletUtils.js]: nit: indent 2 spaces instead of 4 here

File: apps/src/gamelab/dropletConfig.js [hunk 3]
```diff
@@ -23,12 +23,23 @@ var spriteBlockPrefix = 'sprite.';
...
+function genPropExpansion (propname) {
+  return function (block) {
+    if (!block || block.type === 'socket') {
+      return propname;
+    } else {
+      return propname + ' = __;';
+    }
+  };
+}
+
```
Comment (Bjvanminnen) [inline apps/src/gamelab/dropletConfig.js]: Is this something we can/should share? It looks identical between multiple files atm

**Comment by Bjvanminnen** [comment]
lgtm

File: apps/src/gamelab/dropletConfig.js (hunk 3 above)
Comment (cpirich) [inline apps/src/gamelab/dropletConfig.js]: Oh, I meant to delete this and only use the one in `dropletConfig.js` - good catch!

**Pull request merged**